src/compile/data/bin.ts

```diff
@@ -126,7 +128,15 @@ export class BinNode extends DataFlowNode {
   public merge(other: BinNode, model: Model) {
     for (const key of keys(other.bins)) {
       if (key in this.bins) {
-        model.renameSignal(other.bins[key].signal, this.bins[key].signal);
+        if (model) {
```

Comment:
I added this bit for the sole purpose of testing. I wasn't sure how best to test the `merge` method in tests where I created instances of BinNodes since I didn't have access to a model. I suppose I could take the `if` out and just pass an arbitrary model to the method for those tests.

Reply:
An even cleaner way would be to pass in the `renameSinal` method only. In the tests, you can just pass in `(s) -> s` then.

Reply:
That's a much nicer solution. I'll go ahead and implement it.

Reply:
I remember doing this in some other places but must have forgotten it here. Thanks for taking care of it.